src/__tests__/concat-test.js

```diff
@@ -0,0 +1,116 @@
+import concat from '../concat';
+import { List, Map, Seq, Set } from 'immutable';
+
+describe('transmute/concat', () => {
+  it('throws an error when subject is `null` or `undefined`', () => {
+    expect(() => concat(List(), null)).toThrow();
+    expect(() => concat(Seq(), null)).toThrow();
+  });
+
+  it('throws an error when attempting to concatenate non-indexed iterables', () => {
+    expect(() => concat({}, Map())).toThrow();
+    expect(() => concat({}, Set())).toThrow();
+    expect(() => concat({}, Seq.Keyed())).toThrow();
+  });
+
+  it('correctly concatenates values when update is null', () => {
```

Comment:
Is this what we expect? I would expect `concat(null, List())` to either throw or return `List()`

Reply:
This is the default behavior of the Immutable library:
http://neilff.github.io/immutable-repl/?q=Y29uc3QgbGlzdCA9IEltbXV0YWJsZS5MaXN0KFsxLCAyLCAzXSk7CgpsaXN0LmNvbmNhdChudWxsKQ%3D%3D